core/upgrade/upgrade.go

```diff
@@ -29,22 +29,23 @@ const (
 	Error
 )
 
+// States holds all the possible states.
+var States = map[State]string{
```

Comment:
shouldn't this be an enum? we can still iterate over its values on the test, right?

Reply:
Go doesn't have enums. Aside from that, I'd like to know all the valid states available. This just does that.

Reply:
lol you know what I mean! 
```go
type States string
const (
        Created States = "created"
        ...
)
```

Reply:
but indeed, iterating over those is less idiomatic